extensions/wikia/Recirculation/js/recirculation.js

```diff
@@ -0,0 +1,184 @@
+	 *  			any options to pass to the helper
+	 *  		}
+	 *  	}
+	 *  ];
+	 *
+	 */
+	'use strict';
+
+	var recircModules = [
+			{
+				id: "LI_rail",
+				viewModule: 'ext.wikia.recirculation.views.premiumRail',
+				options: {
+					max: 5,
+					widget: 'wikia-rail',
+					source: 'fandom',
+					opts: {
+						resultType: 'cross-domain',
+						domainType: 'fandom.wikia.com'
+					}
+				}
+			},
+			{
+				id: "LI_impactFooter",
+				viewModule: 'ext.wikia.recirculation.views.impactFooter',
+				options: {
+					max: 9,
+					widget: 'wikia-impactfooter',
+					source: 'fandom',
+					opts: {
+						resultType: 'cross-domain',
+						domainType: 'fandom.wikia.com'
+					}
+				}
+			},
+			{
+				id: "LI_footer",
+				viewModule: 'ext.wikia.recirculation.views.footer',
+				options: {
+					max: 3,
+					widget: 'wikia-footer-wiki-rec',
+					source: 'wiki',
+					title: "Discover New Wikis",
+					width: 332,
+					height: 187,
+					flush: true,
+					opts: {
+						resultType: 'subdomain',
+						domainType: 'fandom.wikia.com'
+					}
+				}
+			}
+		],
+		logGroup = 'ext.wikia.recirculation.experiments.mix',
+		views = {}, // Each view holds an array of promises used to gather data for that view
+		completed = [], // An array of promises to keep track of which views have completed rendering
+		liftigniterHelpers = {};
+
+	if (w.wgContentLanguage !== 'en') {
+		if (videosModule) {
+			videosModule('#recirculation-rail');
+		}
+		discussions();
+		return;
+	}
+
+	recircModules.forEach(function (recircModule) {
+		var deferred = $.Deferred();
+
+		var configuredHelper = liftigniter(recircModule.options);
+		configuredHelper.loadData()
+			.done(function (data) {
+				deferred.resolve(data);
+			})
+			.fail(function (err) {
+				deferred.reject(err);
+			});
+
+		// We need to keep track of the liftigniter helpers so we can setup tracking for them
+		liftigniterHelpers[recircModule.id] = configuredHelper;
+
+		if (recircModule.viewModule) {
+			views[recircModule.viewModule] = views[recircModule.viewModule] || [];
+			views[recircModule.viewModule].push(deferred);
+		}
+	});
+
+	$.each(views, function (viewModule, promises) {
+		var deferred = $.Deferred();
+
+		completed.push(deferred);
+
+		log('Initializing View: ' + viewModule, 'info', logGroup);
+		require([viewModule], function (viewFactory) {
+			$.when.apply($, promises)
+				.done(function () {
+					var view = viewFactory(),
+						args = Array.prototype.slice.call(arguments),
+						data = {
+							title: '',
+							items: []
+						};
+
+					log(args, 'info', logGroup);
+
+					args.forEach(function (result) {
+						if (!result) {
+							return;
+						}
+
+						if (result.title && data.title.length === 0) {
+							data.title = result.title;
+						}
+
+						data.items = data.items.concat(result.items);
+					});
+
+					view.render(data)
+						.then(view.setupTracking())
+						.then(deferred.resolve);
+				})
+				.fail(handleError(viewModule), function () {
```

Comment:
Is this code suppose to be an ABTest or is it production implementation?
If we're productizing this feature maybe it makes sense to render some recirc modules when only one promise is rejected?